roles/openshift_health_checker/openshift_checks/ram_diskspace_availability.py

```diff
@@ -0,0 +1,42 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class RamDiskspaceAvailability(NotContainerizedMixin, OpenShiftCheck):
+    """Check that recommended memory and disk-space are available."""
+
+    name = "ram_diskspace_availability"
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        diskfree = self.to_gigabytes(self.get_disk_space(ansible_mounts))
+        memoryfree = get_var(task_vars, "facter_memoryfree")
+
+        recommended_memory = get_var(task_vars, "min_memory_gb")
+        recommended_storage = get_var(task_vars, "min_diskspace_gb")
```

Comment:
You can get the supported values here:
https://docs.openshift.com/container-platform/latest/install_config/install/prerequisites.html#hardware

Even though there may be different requirements for older versions of the product, in general I think we should always encourage customers to meet the requirements of the latest version so they don't hit problems when they upgrade.